homework/grayscale-image/compression.cpp

```diff
@@ -0,0 +1,50 @@
+#include "compression.hpp"
+
+#include <algorithm>
+#include <iostream>
+
+CompressedBitmap compressGrayscale(const Bitmap& bitmap) {
+    CompressedBitmap compressed;
+    compressed.reserve(width * height);
+    int row{};
+    do {
+        for (auto beg = bitmap[row].begin(),
+                  end = bitmap[row].end();
+             beg != end;) {
+            const auto diff = std::find_if(beg, end,
+                                           [&beg](const auto& el) {
+                                               return el != *beg;
+                                           });
+            compressed.emplace_back(*beg, std::distance(beg, diff));
+            beg = diff;
+        }
+    } while (++row != height);
```

Comment:
You can use transform :) or std::generate (you will capture iterator and then jump between different numbers and return proper pair :) )